### PR TITLE
More effect fixes

### LINKED ISSUE #367: Ferry of Styx wrong lightning color and possibly texture
Seems like RGB vs BGR issue, but the texture itself may also be wrong because it looks more like fire.

ferry

### PR DESCRIPTION
**Pull request by theflyingzamboni**
Fixed Effects:

-All or almost all lightning effects (closes #367)
-Screenshot effects (Death Dimension, Melbu attack, Kubila's demon frog)
-Incorrect blending on effects (e.g. red glow effect in Death Dimension)
-Lighting on many sprite effects
-Invisible combat effects (like the explosions in Trans Light)

### CODE REVIEW

File: src/main/java/legend/game/combat/SEffe.java [hunk 1]
```diff
@@ -585,13 +585,14 @@ private SEffe() { }
...
+
   /**
    * <ol start="0">
    *   <li>{@link SEffe#FUN_8010b594}</li>
-   *   <li>{@link SEffe#FUN_8010bc60}</li>
+   *   <li>{@link SEffe#renderScreenCapture}</li>
    * </ol>
    */
-  private static final ArrayRef<Pointer<BiConsumerRef<EffectManagerData6c, DeathDimensionEffect1c>>> deathDimensionRenderers_80119fec = MEMORY.ref(4, 0x80119fecL, ArrayRef.of(Pointer.classFor(BiConsumerRef.classFor(EffectManagerData6c.class, DeathDimensionEffect1c.class)), 2, 4, Pointer.deferred(4, BiConsumerRef::new)));
+  private static final ArrayRef<Pointer<BiConsumerRef<EffectManagerData6c, ScreenCaptureEffect1c>>> screenCaptureRenderers_80119fec = MEMORY.ref(4, 0x80119fecL, ArrayRef.of(Pointer.classFor(BiConsumerRef.classFor(EffectManagerData6c.class, ScreenCaptureEffect1c.class)), 2, 4, Pointer.deferred(4, BiConsumerRef::new)));
```
Comment (Ink230) [inline src/main/java/legend/game/combat/SEffe.java:595]: CS: space before >>>

File: src/main/java/legend/game/tmd/Renderer.java [hunk 2]
```diff
@@ -206,7 +211,7 @@ public static void renderTmdPrimitive(final TmdObjTable1c.Primitive primitive, f
         }
       }
 
-      if(translucent && !textured) {
+      if(translucent && (!textured || uniformLit)) {
```
Comment (Ink230) [inline src/main/java/legend/game/tmd/Renderer.java:214]: if is true if
transculent is true
AND
textured is false or it is uniformLit

so we want translucent true and uniformLit true
or
translucent true and textured false

just ensuring this is the intended logic and not missing any cases. ✔

**Review by Ink230 — APPROVED** [review]
Only one minor CS issue. 🙃
Great renaming. ✔
Humourous bug fixes. ✔

File: src/main/java/legend/game/combat/SEffe.java (hunk 1 above)
Comment (LordMonoxide) [reply src/main/java/legend/game/combat/SEffe.java:595]: These are just the closing brackets for the generic type so this is fine as-is

File: src/main/java/legend/game/combat/effects/TransformScaler6cSub34.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package legend.game.combat.effects;
+
+import legend.core.gte.VECTOR;
+
+public class TransformScaler6cSub34 extends BttlScriptData6cSubBase2 {
```
Comment (LordMonoxide) [inline src/main/java/legend/game/combat/effects/TransformScaler6cSub34.java:5]: "BttlScriptData6c" was the old name for EffectManagerData6c, so this should be renamed to `TransformScalerEffect34`

**Review by LordMonoxide — CHANGES_REQUESTED** [review]

File: src/main/java/legend/game/tmd/Renderer.java (hunk 2 above)
Comment (theflyingzamboni) [reply src/main/java/legend/game/tmd/Renderer.java:214]: Correct. If a translucent poly isn't textured (so just color) then the back lighting should be  always be applied. If it is textured, the lighting should only be applied if the primitive is flagged as having uniform lighting.

File: src/main/java/legend/game/combat/SEffe.java [hunk 4]
```diff
@@ -2806,13 +2808,16 @@ public static void renderGradient(final USCOLOUR colour1, final USCOLOUR colour2
...
-      .rgb(1, colour2.getX(), colour2.getY(), colour2.getZ())
+      .rgb(1, colour2.getX() >>> 8, colour2.getY() >>> 8, colour2.getZ() >>> 8)
       .monochrome(2, 0)
-      .rgb(3, colour1.getX(), colour1.getY(), colour1.getZ());
+      .rgb(3, colour1.getX() >>> 8, colour1.getY() >>> 8, colour1.getZ() >>> 8);
 
     GPU.queueCommand(a3 + a4 >> 2, cmd);
   }
 
+  /* renderer used for many (though not all) lightning effects
```
[the remark this answers is not in the document]
Comment (theflyingzamboni) [reply src/main/java/legend/game/combat/SEffe.java:2818]: Whoops, guess I missed that one.

File: src/main/java/legend/game/tmd/Renderer.java [hunk 5]
```diff
@@ -7,8 +7,11 @@
 import legend.core.gte.GsDOBJ2;
 import legend.core.gte.SVECTOR;
 import legend.core.gte.TmdObjTable1c;
+import legend.game.combat.environment.BattleLightStruct64;
```
Comment (LordMonoxide) [inline src/main/java/legend/game/tmd/Renderer.java:10]: TI don't think any of these new imports are used?

**Review by LordMonoxide — CHANGES_REQUESTED** [review]

File: src/main/java/legend/game/tmd/Renderer.java (hunk 5 above)
Comment (theflyingzamboni) [reply src/main/java/legend/game/tmd/Renderer.java:10]: BattleLightStruct is used for uniform lighting

**Review by LordMonoxide — APPROVED** [review]

**Pull request merged**